Library/Homebrew/utils/bottles.rb

```diff
@@ -59,10 +59,38 @@ def formula_contents(bottle_file,
           name: resolve_formula_names(bottle_file)[0])
         bottle_version = resolve_version bottle_file
         formula_path = "#{name}/#{bottle_version}/.brew/#{name}.rb"
-        contents = Utils.popen_read "tar", "-xOzf", bottle_file, formula_path
+        command = tar_command "xOf", bottle_file, formula_path
+        receipt_file = `#{command}`
         raise BottleFormulaUnavailableError.new(bottle_file, formula_path) unless $CHILD_STATUS.success?
         contents
       end
+
+      def install_dependencies(compression_type, ignore_tar: false)
+        return unless compression_type == :xz
+        return unless ignore_tar || DependencyCollector.tar_needs_xz_dependency?
+        f = Formula["xz"]
+        return if f.optlinked?
+        fi = FormulaInstaller.new(f)
+        fi.installed_as_dependency = true
+        fi.installed_on_request    = false
+        fi.prelude
+        oh1 "Installing bottle dependency: #{Formatter.identifier(f.name)}"
+        fi.install
+        fi.finish
+      end
+
+      private
+
+      def xzpath
+        "#{HOMEBREW_PREFIX}/opt/xz/bin/xz"
```

Comment:
I'd just use a blanket xz here and ensure it was added to the PATH previously if needed.